airgun/entities/ansible_role.py

```diff
@@ -0,0 +1,75 @@
+from navmazing import NavigateToSibling
+
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep
+from airgun.navigation import navigator
+from airgun.views.ansible_role import AnsibleRolesImportView
+from airgun.views.ansible_role import AnsibleRolesView
+
+
+class AnsibleRolesEntity(BaseEntity):
+    """Main Ansible roles entity"""
+
+    endpoint_path = '/ansible/ansible_roles'
+
+    def search(self, value):
+        """Search for existing Ansible Role"""
+        view = self.navigate_to(self, 'All')
+        view.search(value)
+        return view.table.read()
+
+    def delete(self, entity_name):
+        """Delete Ansible Role from Satellite"""
+        # This method currently does not work as the Name column table header
+        # cell includes the `▲` character used for sorting that column
+        view = self.navigate_to(self, 'All')
+        view.search(entity_name)
+        view.table.row()['Actions'].widget.fill('Delete')
+        view.confirm.confirm_dialog.click()
```

Comment:
You should be able to use `view.dialog.confirm()` here, as the PF4 confirmation dialog modal is on the BaseView as `dialog`.